apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js

```diff
@@ -0,0 +1,187 @@
+/** @file Playground Component setup tests */
+import five from 'johnny-five';
+import Playground from 'playground-io';
+import {expect} from '../../../../util/configuredChai';
+import sinon from 'sinon';
+import {
+  initializeAccelerometer,
+  initializeButton,
+  initializeColorLeds,
+  initializeLightSensor,
+  initializeSoundSensor,
+  initializeThermometer,
+  initializeTouchPads,
+} from '@cdo/apps/lib/kits/maker/PlaygroundComponents';
+import TouchSensor from '@cdo/apps/lib/kits/maker/TouchSensor';
+
+// Polyfill node's process.hrtime for the browser, gets used by johnny-five.
+process.hrtime = require('browser-process-hrtime');
+
+describe('Circuit Playground Components', () => {
+  let board;
+
+  beforeEach(() => {
+    board = newBoard();
+  });
+
+  describe('initializeColorLeds()', () => {
+    it('initializes all ten LEDs', () => {
+      const components = initializeColorLeds(board);
+
+      // Return ten controllers
+      expect(Object.keys(components)).to.deep.equal([
+        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'
+      ]);
+
+      // Every returned controller is a five.Led.RGB
+      Object.keys(components).forEach(key => {
+        expect(components[key]).to.be.an.instanceOf(five.Led.RGB);
+      });
+
+      // 20 sysex calls during initialization, two per LED.
+      for (let i = 0; i < 10; i++) {
+        // TODO (bbuchanan): Record what these calls mean.
+        expect(board.io.sysexCommand.getCall(i*2)).to.have.been.calledWith([0x40, 0x10, i, 0, 0, 0, 0]);
+        expect(board.io.sysexCommand.getCall(i*2+1)).to.have.been.calledWith([0x40, 0x11]);
+      }
+    });
+
+    it('can initialize a second set of leds with a second board', () => {
+      // Checks a necessary condition for a true johnny-five level reset.
+      const boardOne = newBoard();
+      const controllers = initializeColorLeds(boardOne);
+      expect(controllers[0].board === boardOne).to.be.true;
+      expect(boardOne.io.sysexCommand.callCount).to.equal(20);
+
+      const boardTwo = newBoard();
+      const newControllers = initializeColorLeds(boardTwo);
+      expect(newControllers[0].board === boardTwo).to.be.true;
+      expect(boardTwo.io.sysexCommand.callCount).to.equal(20);
+    });
+  });
+
+  describe('initializeSoundSensor()', () => {
+    it('initializes one sensor', () => {
+      const sensor = initializeSoundSensor(board);
+      expect(sensor).to.be.an.instanceOf(five.Sensor);
+      // Doesn't use sysex at first
+      expect(board.io.sysexCommand.callCount).to.equal(0);
+    });
+
+    it('gets sensor methods', () => {
+      const sensor = initializeSoundSensor(board);
+      expect(sensor).to.haveOwnProperty('start');
+      expect(sensor).to.haveOwnProperty('getAveragedValue');
+      expect(sensor).to.haveOwnProperty('setScale');
```

Comment:
These particular methods are not part of `five.Sensor` - we monkeypatch them on with [`addSensorFeatures()`](https://github.com/code-dot-org/code-dot-org/blob/e4720aaa135a10fa5b8c9d18d65de913495a884b/apps/src/lib/kits/maker/PlaygroundComponents.js#L107) to create the API our curriculum team requested.